integration-tests/shard-operator-integration-tests/shard-operator-integration-tests-v2/src/test/java/com/redhat/service/smartevents/shard/operator/v2/cucumber/steps/BridgeResourceSteps.java

```diff
@@ -0,0 +1,75 @@
+package com.redhat.service.smartevents.shard.operator.v2.cucumber.steps;
+
+import java.io.ByteArrayInputStream;
+import java.io.InputStream;
+import java.nio.charset.StandardCharsets;
+import java.time.Duration;
+import java.util.Base64;
+import java.util.HashMap;
+import java.util.Map;
+
+import org.awaitility.Awaitility;
+
+import com.redhat.service.smartevents.integration.tests.context.TestContext;
+import com.redhat.service.smartevents.integration.tests.context.resolver.ContextResolver;
+import com.redhat.service.smartevents.shard.operator.v2.resources.ManagedBridge;
+
+import io.cucumber.java.en.And;
+import io.cucumber.java.en.When;
+import io.fabric8.kubernetes.api.model.ObjectMetaBuilder;
+import io.fabric8.kubernetes.api.model.Secret;
+import io.fabric8.kubernetes.api.model.SecretBuilder;
+import io.fabric8.kubernetes.api.model.networking.v1.Ingress;
+
+public class BridgeResourceSteps {
+
+    private TestContext context;
+
+    public BridgeResourceSteps(TestContext context) {
+        this.context = context;
+    }
+
+    @When("^deploy BridgeResource \"([^\"]*)\" using topic \"([^\"]*)\":$")
+    public void deployBridgeResource(String bridgeName, String topic, String bridgeResourceYaml) throws InterruptedException {
+        Map<String, String> secretData = new HashMap<>();
+        addSecretPair(secretData, "bootstrap.servers", System.getProperty("it.shard.kafka.bootstrap.servers"));
+        addSecretPair(secretData, "user", System.getProperty("it.shard.kafka.user"));
+        addSecretPair(secretData, "password", System.getProperty("it.shard.kafka.password"));
+        addSecretPair(secretData, "protocol", "SASL_SSL");
+        addSecretPair(secretData, "topic.name", context.getKafkaTopic(topic));
+        addSecretPair(secretData, "sasl.mechanism", "PLAIN");
+        Secret secret = new SecretBuilder()
+                .withMetadata(
+                        new ObjectMetaBuilder()
+                                .withNamespace(context.getNamespace())
+                                .withName(bridgeName)
+                                .build())
+                .withData(secretData)
+                .build();
+        context.getClient().secrets().inNamespace(context.getNamespace()).withName(bridgeName).createOrReplace(secret);
```

Comment:
I think it could be better to create secret as separate step. That way the scenario will be more explicit, showing what actually the implementation do.

Reply:
I'm not sure about this. In the future the config map should be generated automatically by the operator. Currently the creation of config map is temporary workaround. Therefore it is now closely related to ManagedBridge.

Reply:
The secret is actually created by "synchronizer" part of our operator. So it is sort of "generated automatically".
As in this case we test just the reconciler part of our operator we need to simulate all the synchronizer actions.

As synchronizer creates both CR and secret then we should create both.

In current approach the creation of secret is hidden in test implementation. As a result our tests are not explicit as I would hope.

Reply:
@sutaakar Thank you for the explanation! I've created new step for secret.